help: Update "Star a message" and document mobile features.

drrosa opened the pull request:
- Updates [Star a message](https://zulip.com/help/star-a-message) mobile instructions.
- Updates "View your starred messages" section.
- Adds instructions block and relative link to Starred messages.
- Adds "Toggle starred messages counter" subheading.
-  Documents "Starred messages" mobile feature.
- Adds "Searching for messages" as a related article.

**Screenshots and screen captures:**

![image](https://github.com/zulip/zulip/assets/2343554/a18da506-44d1-4be9-8ef6-b6d306675ef4)

![image](https://github.com/zulip/zulip/assets/2343554/b6c141e2-c1ac-473d-acb4-4868bf300e67)

<img width="250" alt="image" src="https://github.com/zulip/zulip/assets/2343554/3cbad16f-0361-4d8b-882e-84df645e6c45">

<details>
<summary>Self-review checklist</summary>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Automated tests.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Links.
</details>

drrosa commented:
@laurynmm This PR is ready for review. 
@zulipbot add "help center review" "area: mobile"

laurynmm commented:
@drrosa - This looks great! My only thought is maybe adding "Searching for messages" as a related article since `is:starred` can be used as a search filter?

drrosa commented:
> My only thought is maybe adding "Searching for messages" as a related article since `is:starred` can be used as a search filter?

Yeah, good thinking. I updated the PR.

drrosa commented:
> My only thought is maybe adding "Searching for messages" as a related article since `is:starred` can be used as a search filter?

Yeah, good idea. I updated the PR and added a screenshot.

timabbott commented:
Merged, thanks @drrosa and @laurynmm!
